save-backend/src/main/kotlin/org/cqfn/save/backend/security/ProjectPermissionEvaluator.kt

```diff
@@ -36,6 +41,10 @@ class ProjectPermissionEvaluator {
 
     private fun Authentication.hasRole(role: Role): Boolean = authorities.any { it.authority == role.asSpringSecurityRole() }
 
-    private fun hasWriteAccess(userId: Long?, project: Project) =
-            userId != null && (project.userId == userId || userId in project.adminIdList())
+    private fun hasWriteAccess(userId: Long?, project: Project): Boolean = if (userId != null && project.userId == userId) {
+        true
+    } else {
+        val adminIdList = lnkUserProjectService.getAllUsersByProjectAndRole(project, Role.ADMIN).map { it.id }
```

Comment:
We should probably consider caching such thing in the future (#575), because permissions don't change often and Hibernate's L2 cache can spare us a lot of DB queries here.

Reply:
Yes I agree. I think this should be a issue